Brass monkey

ironsnake345 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This is a redux of my old brass monkey addition, since TG's code had some fuckery that moved it to its new repository. This PR does two things: adds Brass Monkey, a drink crafted by mixing Kong with iron, copper and banana juice, and updates monkey aggression code to check for mobs with the Jungle faction as non-eligible targets for monkey aggression, instead of just checking for other monkeys as non-eligible. Now you have a highly alcoholic drink that makes monkeys not wanna kill you, and monkeys won't attack other jungle mobs. Yay!
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
It's a silly addition, which also causes monkey's aggression code to make a bit more thematic sense. Who doesn't love silly, memey, far-flung jokes about very old and criminally catchy pop songs? Besides, spess is a scary place. A good spessman always needs a new way to get drunk off his ass.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Added Brass Monkey, an alcoholic drink mixed by combining 5u Kong, 5u Banana juice, 1u copper, and 1u iron. It makes 10 units of a drink that makes monkeys very hesitant to try and kill you while it's in your system! Who could assault such a funky monkey as you?
tweak: Monkeys are now jungle mobs
balance: Monkeys are now jungle mobs
imageadd: added a sprite for a glass of Brass Monkey
code: Monkey aggression code now decides whether or not they want to attack something based off of whether or not it is a jungle mob or has the Monkeybro trait, not whether or not it's another monkey.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

ironsnake345 committed:
Adds Brass Monkey, a new mixable drink

Brass monkey can be mixed from 5 parts kong, 5 parts banana juice, 1 part copper, 1 part iron. (makes 10 units instead of 12)

ironsnake345 committed:
Update alcohol_reagents.dm

ironsnake345 committed:
Update code/modules/food_and_drinks/recipes/drinks_recipes.dm

forgot the newline, sorry. Thanks, Tivi

Co-Authored-By: TiviPlus <57223640+TiviPlus@users.noreply.github.com>

ironsnake345 committed:
Update alcohol_reagents.dm

Capitalized glass description, added newline

ironsnake345 committed:
Merge branch 'Brass-Monkey' of https://github.com/ironsnake345/tgstation into Brass-Monkey

ironsnake345 committed:
Update code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

Co-Authored-By: spookydonut <github@spooksoftware.com>

ironsnake345 committed:
Changed brass monkey to give the jungle faction while it's in your system (also minor rework to monkey aggression code so that they don't wanna hurt you if you're a jungle guy)

ironsnake345 committed:
Merge branch 'master' into Brass-Monkey

ironsnake345 committed:
Aggression fix

Fixed monkey de-aggression, and that thing Cobby pointed out

ironsnake345 committed:
Added an ! that I forgot

ATH1909 commented:
Is it intentional that people without livers can retain the effects of this chem indefinitely?

Did the microdosing bugs related to on_mob_delete() get fixed? If not, you might want to try using on_mob_metabolize() and on_mob_stop_metabolize() instead of on_mob_add() and on_mob_delete().

ironsnake345 commented:
Good idea, ATH! I hadn't considered that possibility. Plus, I'm pretty new to the whole TG coding thing tbh, so advice like that is welcome.

ironsnake345 committed:
Now respects liver function

ATH1909 commented on code/modules/mob/living/carbon/monkey/combat.dm:
```diff
@@ -123,8 +123,8 @@
-	// target non-monkey mobs when aggressive, with a small probability of monkey v monkey
-	if(aggressive && (!istype(L, /mob/living/carbon/monkey/) || prob(MONKEY_AGGRESSIVE_MVM_PROB)))
+	// target non-jungle mobs when aggressive, with a small probability of monkey v monkey. Also avoids attacking  monkey bros.
+	if(aggressive && (!faction_check_mob(L, FALSE) || !HAS_TRAIT(L, TRAIT_MONKEYBRO) || prob(MONKEY_AGGRESSIVE_MVM_PROB)))
```
```suggestion
	if(aggressive && (!(faction_check_mob(L, FALSE) || HAS_TRAIT(L, TRAIT_MONKEYBRO)) || prob(MONKEY_AGGRESSIVE_MVM_PROB)))
```

ironsnake345 replied:
Oh, good idea, I like that. Committing!

Anonmare commented:
Cold enough to freeze the balls off a brass monkey

Qustinnus commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2239,3 +2239,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	glass_desc = "Coolin' by the lockers, getting kind of funky, me and the crew, we're drinking Brass Monkey."
+
+/datum/reagent/consumable/ethanol/brass_monkey/on_mob_metabolize(mob/living/M)
+	if(!HAS_TRAIT_FROM(M, TRAIT_MONKEYBRO, BRASS_MONKEY))
```
i dont think you have to check this it does this on its own but dont quote me on that

Qustinnus commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2239,3 +2239,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
+		ADD_TRAIT(M, TRAIT_MONKEYBRO, BRASS_MONKEY)
+
+/datum/reagent/consumable/ethanol/brass_monkey/on_mob_end_metabolize(mob/living/M)
+	if(HAS_TRAIT_FROM(M, TRAIT_MONKEYBRO, BRASS_MONKEY))
```
ditto

ironsnake345 replied:
You're probably right, but I want to go on the safe side of things. Besides, I seriously doubt one HAS_TRAIT_FROM proc whenever someone sips this drink will have a noticeable effect on performance.

Qustinnus replied:
that's not a good argument. on its own barely any check will really dent performance on its own.  It's the accumulative dent tons of those checks would have if noone cared that would affect performance, though.

@Rohesie you worked with the trait code, does it matter if you remove / add traits without checking first? I'd imagine not right?

Rohesie replied:
This is very much a no.
This check is superfluous at best and error-hiding at worst.
Use `ADD_TRAIT(M, TRAIT_MONKEYBRO, type)` and `REMOVE_TRAIT(M, TRAIT_MONKEYBRO, type)`
No need to define a new string for a reagent, the type is the convention.

ironsnake345 committed:
Update code/modules/mob/living/carbon/monkey/combat.dm

Used parentheses to reduce the number of nots in the code that determines if a monkey should or should not aggro to something.

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

ExcessiveUseOfCobblestone commented:
just re-pr when you do reviews and fix conflicts.
